files/files/main.cpp

```diff
@@ -0,0 +1,29 @@
+// files.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include "pch.h"
+#include "directory.h"
+#include "Myfile.h"
+using  namespace std;
+namespace fs = std::filesystem;
+
+int main()
+{
+
+	string path = "C:\\Users\\Anastasia Antonyk\\Desktop\\SoftserveITATasks";
+	cout << "Path:\t" << path << endl;
+	cout << "Number of detected files:\t" << count_cpp_files(path) << endl;
+	vector<fs::path> cppfiles = list_cpp_files_in_directory(path); //find cpp files in chosen directory
+	vector<Myfile> files = objects(cppfiles); //making vector of object of cppfiles
+	vector<thread> t;
+	for (auto p : files) {
+		t.emplace_back(my_thread, p);
```

Comment:
It might be a bad idea to create a vector for each thread. When we create more threads than our operation system can support (more then the cores number) our physical thread will switch any time thread's function will work. It can be time consuming  to make this switch operation. I'd suggest to create only the number of threds which our system can support (FE 4) and run any task to this thread when it's don't have any work to do.